Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs

```diff
@@ -293,5 +298,26 @@ void HandleOrThrow(Action action)
         }
 
         #endregion
+
+        private void UserId_KeyDown(object sender, System.Windows.Input.KeyEventArgs e)
+        {
+            if (e.Key == Key.Return)
+            {
+                var userId = UserId.Text;
+                var text = string.IsNullOrEmpty(userId) ? null : userId;
+                AppCenter.SetUserId(text);
+                Settings.Default.UserId = text;
+                Settings.Default.Save();
```

Comment:
Those lines are the exact same as UserId_lostFocus, should use a private method for both

Reply:
Make sense. Added a private method.